src/main/ruby/truffleruby/core/array.rb

```diff
@@ -296,11 +296,45 @@ def dig(idx, *more)
     if Primitive.nil?(result) || more.empty?
       result
     else
-      raise TypeError, "#{result.class} does not have #dig method" unless result.respond_to?(:dig)
-      result.dig(*more)
+      Array.looping_dig(result, *more)
     end
   end
 
+  def self.looping_dig(obj, *idxs)
```

Comment:
Is this any faster than the previous approach?
I think the previous approach is more elegant.

I guess the main difference is avoiding extra sub-arrays, but those are already copy-on-write and potentially escape-analyzed so on peak shouldn't be too bad.

Reply:
We think you get a performance cliff when inlining gives up. Also faster in the interpreter. We'll keep exploring and will include benchmarks in the final PR.